cooked-validators/src/Cooked/MockChain/Monad.hs

```diff
@@ -64,6 +65,12 @@ class (MonadFail m) => MonadBlockChain m where
     UtxoPredicate a ->
     m [(SpendableOut, Maybe a)]
 
+  utxosSuchThisAndThat ::
+    (Pl.FromData a) =>
+    (Pl.Address -> Bool) ->
+    UtxoPredicate a ->
+    m [(SpendableOut, Maybe a)]
```

Comment:
A cleaner way to do this would be to modify the currently existing `utxosSuchThat` to have a predicate that takes three arguments (datum, value, and address), instead of introducing this additional function (with a somewhat odd name). However, this would be a breaking change, and we need to discuss if this is what we want to do.

Reply:
I think the implementation of the class for `Contract` doesn't support this function. I actually had a comment along the lines of "why not unite the two", but after seeing this function implemented as `undefined`/`error` here I removed it.

Reply:
That's true, I forgot about that part. Still, I'm not entirely sure what the point of Contract is anyway, and if it is that much of an issue if Contract doesn't support it